kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -1265,6 +1332,31 @@ class ResidualBasedNewtonRaphsonStrategy
 
     bool mKeepSystemConstantDuringIterations; // Flag to allow keeping system matrix constant during iterations
 
+    /**
+     * @brief This vector stores the non-converged solutions at each iteration
+     * @details Each entry in the vector represents the solution vector at a given iteration which did not meet the convergence criteria.
+     */
+    std::vector<Vector> NonconvergedSolutions;
```

Comment:
do not make this as a class variable. Anyway you are deallocating (by a clear) at the end of the SolveSolutionStep so define it locally

Reply:
It is now a local varible